mypy/util.py

```diff
@@ -134,3 +140,36 @@ def id(self, o: object) -> int:
             self.id_map[o] = self.next_id
             self.next_id += 1
         return self.id_map[o]
+
+
+if sys.version_info >= (3, 6):
+    PathType = Union[AnyStr, os.PathLike]
+else:
+    PathType = AnyStr
+
+
+def _replace(src: PathType, dest: PathType, timeout: float = 10) -> None:
+    '''
+    Replace src with dest using os.replace(src, dest)
+    Wait and retry if OSError exception is raised
+    Increase wait time exponentially until total wait of timeout sec
+    On timeout, give up and reraise the last exception seen
+    '''
+    n_iter = max(1, math.ceil(math.log2(timeout / 0.001)))
```

Comment:
Well, I need to know what `wait` should start from. I'd like to start it from somewhere around 0.001 sec, but of course I don't care about the precise value. I do care about the total timeout though, which means I care about the *final* (rather than *starting*) value of the wait. Specifically, if my timeout is say 1 sec, I want my final `wait` just before I give up, to be 0.5 sec (that way the total wait will be `0.5 + 0.5/(2**1) + 0.5/(2**2) + ... = 1`).

But in order to get the correct starting value without math, I'd need to create a second loop, where I repeat `timeout /= 2` until it gets to (say) below 0.001. I can do that, but is it better than calling `math.log`? 

Actually, it might be easier to read, let me commit such a version, and let's decide which is better.